qadi_simple.py

```
@@ -353,10 +365,55 @@
+                        pass
+                
+                # Run evolution with timeout protection
                 evolution_start = time.time()
-                evolution_result = await ga.evolve(request)
-                evolution_time = time.time() - evolution_start
+                progress_task = asyncio.create_task(show_progress(evolution_start, evolution_timeout))
+                
+                try:
+                    evolution_result = await asyncio.wait_for(
+                        ga.evolve(request),
+                        timeout=evolution_timeout
```

Ruff (0.12.2)

Trailing comma missing

Add trailing comma

(COM812)